### PR TITLE
2.0 alpha

### PR DESCRIPTION
**Pull request by simasgrilo**
Pull request with visualization of dataflow and files of a specific trial directly from "now dataflow".

### CODE REVIEW

File: capture/noworkflow/now/vis/views.py [hunk 1]
```diff
@@ -8,13 +8,15 @@
...
 
-from flask import render_template, jsonify, request
+from flask import render_template, jsonify, request, make_response, send_file
 
 from ..persistence.models import Trial
 from ..models.history import History
 from ..models.diff import Diff
 from ..persistence import relational
 
+import subprocess #modulo p chamada de script bash
```
Comment (JoaoFelipe) [inline capture/noworkflow/now/vis/views.py]: Remove comments

File: capture/noworkflow/now/vis/views.py [hunk 2]
```diff
@@ -77,6 +79,39 @@ def trials():
...
+#generate dafalowdataflow
+@app.route("/trials/<tid>/flow.pdf")
+def dataflow(tid):
+    """Generates the dafalow of a trial """ 
+    trial = Trial(tid)
+    dest = 'flow.pdf'
+    temp = 'flow.dot'
+    # args = ['/bin/sh',"/usr/local/lib/python3.5/dist-packages/noworkflow-2.0.0a0-py3.5.egg/noworkflow/now/vis/static/mydataflow.sh",os.getcwd(),str(tid),temp,dest] 
+    #ret = subprocess.Popen(args)
+    cmd = 'cd ' + os.getcwd()+ ' && now dataflow ' + str(tid) + ' >' + temp +' && dot -T pdf ' + temp + ' -o ' + dest
```
Comment (JoaoFelipe) [inline capture/noworkflow/now/vis/views.py]: Either call a function from noWorkflow, or invoke `now` using `sys.executable`. Using the system `now` may invoke the wrong installation.

File: capture/noworkflow/now/vis/views.py [hunk 3]
```diff
@@ -77,6 +79,39 @@ def trials():
...
+    subprocess.run(cmd, check=True, shell=True)      
+    return send_file(os.getcwd() + '/' + dest,attachment_filename=dest)
+
+@app.route("/trials/<tid>/<scriptHash>/<name>")    
+def get_script(tid,scriptHash,name):
+    """Returns the executed script"""
+    dirHash = scriptHash[0] + scriptHash[1]
+    objHash = scriptHash[2::]
+    dir = os.getcwd() + '/.noworkflow/content/' + dirHash + '/' + objHash
```
Comment (JoaoFelipe) [inline capture/noworkflow/now/vis/views.py]: Invoke the content database get function to support the git integration

File: capture/noworkflow/now/vis/views.py [hunk 4]
```diff
@@ -77,6 +79,39 @@ def trials():
...
+
+@app.route("/trials/files/<fileHash>/<fileExt>")
+def get_file(fileHash, fileExt):
+    """Returns a file used in the trial"""
+    #filename: necessário p/ saber a extensão do arquivo.
+    #filehash: p pegar o arquivo no diretório do noworkflow
+    dirHash = fileHash[0] + fileHash[1]
+    objHash = fileHash[2::]
+    #extension = fileName[(len(fileName) - 4)::]
+    dir = os.getcwd() + '/.noworkflow/content/' + dirHash + '/' + objHash
```
Comment (JoaoFelipe) [inline capture/noworkflow/now/vis/views.py]: Invoke the content database get function to support the git integration

File: capture/noworkflow/now/vis/views.py [hunk 5]
```diff
@@ -174,6 +209,7 @@ def diff_accesses(trial1, trial2):
 @app.route("/diff/<trial1>/<trial2>/<graph_mode>-<cache>.json")
 def diff_graph(trial1, trial2, graph_mode, cache):
     """Respond trial diff as JSON"""
+    print("private dancer")
```
Comment (JoaoFelipe) [inline capture/noworkflow/now/vis/views.py]: Hahaha

File: npm/packages/history/src/graph.ts [hunk 6]
```diff
@@ -181,7 +181,7 @@ class HistoryGraph {
       });
     form.append("label")
       .attr("for", "history-" + this.graphId + "-toolbar-fonts")
-      .attr("title", "Set font size")
+      .attr("title", "Set font sizetete")
```
Comment (JoaoFelipe) [inline npm/packages/history/src/graph.ts]: Fix this line

File: npm/packages/history/src/graph.ts [hunk 7]
```diff
@@ -380,6 +380,7 @@ class HistoryGraph {
   selectTrial(trialId: string) {
     for (var node of this.nodes) {
       if (node.title == trialId) {
+        console.log("scarface");
```
Comment (JoaoFelipe) [inline npm/packages/history/src/graph.ts]: ...

File: npm/packages/trial/src/graph.ts [hunk 8]
```diff
@@ -35,6 +35,7 @@ import {
   TrialEdgeData
 } from './structures';
 
+//import $ = require("jquery");
```
Comment (JoaoFelipe) [inline npm/packages/trial/src/graph.ts]: remove

File: npm/packages/trial/src/graph.ts [hunk 9]
```diff
@@ -840,4 +854,11 @@ class TrialGraph {
   private _graphId(): string {
     return "trial-graph-" + this.graphId;
   }
+
+  //dataflow:
+
+  generateDataflow() {
+    return 1;
```
Comment (JoaoFelipe) [inline npm/packages/trial/src/graph.ts]: Check what does it do

File: capture/noworkflow/now/vis/static/mydataflow.sh [hunk 10]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
```
Comment (JoaoFelipe) [inline capture/noworkflow/now/vis/static/mydataflow.sh]: Is this file used?

File: npm/packages/trial/src/graph.ts (hunk 9 above)
Comment (simasgrilo) [reply npm/packages/trial/src/graph.ts]: This should also be excluded. It was a first attempt to implement the file access, but this was in the wrong file.

File: capture/noworkflow/now/vis/static/mydataflow.sh (hunk 10 above)
Comment (simasgrilo) [reply capture/noworkflow/now/vis/static/mydataflow.sh]: nope, @Daniel-02 forgot to delete it after changing how noworkflow is called in the backend

**Pull request merged**